src/keygen/Avvm.hs

```diff
@@ -86,9 +85,9 @@ genGenesis avvm genCerts holder = GenesisData
 getHolderId :: Maybe FilePath -> IO StakeholderId
 getHolderId holderPath = case holderPath of
     Just fileName -> do
-        sk <- maybe
-            (fail "No secret key is found in file")
-            pure =<< view usPrimKey <$> readUserSecret fileName
+        mSk <- view usPrimKey <$> readUserSecret fileName
+        sk <- maybe (fail "No secret key is found in file")
+              pure mSk
```

Comment:
Though i have no idea why you prefere `fail` here. `MonadFail` is a cancer anyway.